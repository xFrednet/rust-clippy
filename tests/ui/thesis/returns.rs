fn simple_const() -> u32 {
    15
}

#[forbid(clippy::borrow_pats)]
fn fn_call() -> u32 {
    simple_const()
}

fn simple_cond() -> u32 {
    if true { 1 } else { 2 }
}

fn static_slice() -> &'static [u32] {
    &[]
}

fn main() {}
