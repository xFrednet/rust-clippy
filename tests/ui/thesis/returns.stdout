# "main"
- Return (-, -, -, -, -) {Const, Unit, AllConst}

# "simple_const"
- Return (-, -, -, -, -) {Const, AllConst}

# "fn_call"
- Return (-, -, -, -, -) {}

# "simple_cond"
- Return (-, -, -, -, -) {Const, Conditional, AllConst}

# "static_slice"
- Return (-, -, -, -, -) {Const, AllConst}

# "static_string"
- Return (-, -, -, -, -) {Const, AllConst}

# "arg_or_default"
- Return (-, -, -, -, -) {Const, Conditional}

